### PR TITLE
GitHub Action generated with haskell-ci

### PR DESCRIPTION
**Pull request by peterbecich**
This is a GitHub Action for Cabal generated by https://github.com/haskell-CI/haskell-ci

The PR includes `cabal.project` to test all three projects

Test:
https://github.com/peterbecich/haskell-kubernetes-client/actions/runs/3683136262

------------

GHC 8.10.7 fails
GHC 9.0.2 succeeds

### CODE REVIEW

File: .github/workflows/haskell-ci.yml [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+    runs-on: ubuntu-18.04
+    timeout-minutes:
+      60
+    container:
+      image: buildpack-deps:bionic
+    continue-on-error: ${{ matrix.allow-failure }}
+    strategy:
+      matrix:
+        include:
+          - compiler: ghc-9.0.2
```
Comment (thomasjm) [inline .github/workflows/haskell-ci.yml:31]: I'd suggest testing GHC 9.2 as well at this point.

Comment (peterbecich) [reply .github/workflows/haskell-ci.yml:31]: There is a compilation error with 9.2:
```
Error: cabal: Could not resolve dependencies:
```

**Comment by thomasjm** [comment]
@peterbecich I'm not sure why you're opening this against my fork, as I'm not the maintainer :P. The real repo is here: https://github.com/kubernetes-client/haskell/

FWIW, GitHub actions configs for Haskell can be much less verbose if you just use the Haskell setup actions that exist nowadays. For example: https://github.com/codedownio/sandwich/blob/master/.github/workflows/sandwich.yml

**Comment by peterbecich** [comment]
@thomasjm , I think https://github.com/kubernetes-client/haskell/ currently only builds with GHC 8. So I was hoping to include this with your PR which fixes the build for GHC 9 in one step

Good point, thanks, that is more concise, I wonder if there are still any advantages to using the generator https://github.com/haskell-CI/haskell-ci

**Comment by thomasjm** [comment]
I'm already having trouble getting things merged over there haha, so making the PR larger isn't likely to help.

I think you should probably cut out the middleman and open the PR directly against the main repo, where I'd be happy to review it.

I've tested many projects with the Haskell setup actions and I think they work great.

**Comment by peterbecich** [comment]
excellent points, thanks @thomasjm

**Comment by peterbecich** [comment]
https://github.com/kubernetes-client/haskell/pull/97

**Pull request closed without merging**